modules/table/src/main/java/org/apache/ignite/internal/table/distributed/storage/InternalTableImpl.java

```diff
@@ -2231,19 +2233,13 @@ private ReplicaRequest upsertAllInternal(
      * @param e Exception to check.
      * @return True if retrying is possible, false otherwise.
      */
-    private boolean isRestartTransactionPossible(Throwable e) {
-        if (e instanceof LockException) {
-            return true;
-        } else if (e instanceof TransactionException && e.getCause() instanceof LockException) {
-            return true;
-        } else if (e instanceof CompletionException && e.getCause() instanceof LockException) {
-            return true;
-        } else if (e instanceof CompletionException
-                && e.getCause() instanceof TransactionException
-                && e.getCause().getCause() instanceof LockException) {
-            return true;
+    private static boolean isRestartTransactionPossible(Throwable e) {
```

Comment:
Method is confusing. It's not possible to restart the transaction just because there was a lock exception or PrimaryReplicaMiss. It's just on of prerequisites.

Reply:
renamed to `exceptionAllowsTxRetry`